### PR TITLE
Convert package to pure ESM + cleanup

### PR DESCRIPTION
**Pull request by fregante**
Closes https://github.com/fregante/github-url-detection/issues/72

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,7 +45,10 @@
 					"*.ts"
 				],
 				"rules": {
-					"@typescript-eslint/no-unnecessary-condition": "error",
+					"ava/assertion-arguments": "off",
```
Comment (fregante) [inline package.json:52]: `strip-indent` is not ESM and the whole TS/ESM setup makes it impossible to get its types without using `allowSyntheticDefaultImports` in `tsconfig`, which can create a broken ESM module.

File: package.json [hunk 2]
```diff
@@ -50,7 +45,10 @@
...
+					"@typescript-eslint/dot-notation": "off",
```
Comment (fregante) [inline package.json:53]: Wait for https://github.com/xojs/eslint-config-xo-typescript/issues/34

File: package.json [hunk 3]
```diff
@@ -50,7 +45,10 @@
...
+					"@typescript-eslint/no-non-null-assertion": "off",
+					"@typescript-eslint/no-non-null-asserted-optional-chain": "off",
```
Comment (fregante) [inline package.json:55]: Unfortunate but most tests use `regex.test()` and it won't be a problem if the result is `regex.test(undefined)`

File: package.json [hunk 4]
```diff
@@ -71,37 +69,41 @@
...
+			"--loader=ts-node/esm",
+			"--experimental-specifier-resolution=node"
 		],
-		"require": [
-			"esm",
-			"ts-node/register"
-		]
+		"nonSemVerExperiments": {
+			"configurableModuleFormat": true
+		}
```
Comment (fregante) [inline package.json:88]: Copied from https://github.com/sindresorhus/refined-github/blob/main/package.json

**Comment by fregante** [comment]
@yakov116 this is a breaking change, so if you have any breaking changes in mind, I'll wait before releasing it. It can stay here a week anyway

**Comment by yakov116** [comment]
Not that I can think of.

Maybe drop `isQuickPR`? I don't see it working, unless you can give me a working link.

One minor thing needs to be updated 
```
// `getRepositoryInfo` depends on `isRepo` and `isRepo` depends on `isRepoSearch`
```

`getRepositoryInfo` does not exist, it was renamed to `isRepo`.

**Comment by fregante** [comment]
> `getRepositoryInfo` does not exist

Sure it does:

https://github.com/fregante/github-url-detection/blob/0b89413b619f54d1cc6264ed78009bceb24e5cff/index.ts#L624

Internally we use `getRepo` for shortness, but we still export it as `getRepositoryInfo`

> Not that I can think of.



> Maybe drop `isQuickPR`? I don't see it working, unless you can give me a working link.

Example URLs are in the code. You can reach that page by visiting a random repository, editing a file, and clicking "Propose file change". The following page will be a compare page with the `quick_pull` parameter

**Comment by fregante** [comment]
We do have some deprecated methods though 👌

File: package.json (hunk 1 above)
Comment (yakov116) [reply package.json:52]: It is now https://github.com/sindresorhus/strip-indent/releases/tag/v4.0.0

**Pull request merged**